src/client/eppo-client.ts

```diff
@@ -372,147 +309,89 @@ export default class EppoClient implements IEppoClient {
     flagKey: string,
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     subjectAttributes: Record<string, any> = {},
+    defaultValue: EppoValue,
     assignmentHooks: IAssignmentHooks | undefined,
     obfuscated: boolean,
-    valueType?: ValueType,
+    expectedVariationType: VariationType,
   ): EppoValue {
-    const { allocationKey, assignment, holdoutKey, holdoutVariation } = this.getAssignmentInternal(
-      subjectKey,
-      flagKey,
-      subjectAttributes,
-      assignmentHooks,
-      obfuscated,
-      valueType,
-    );
-    assignmentHooks?.onPostAssignment(flagKey, subjectKey, assignment, allocationKey);
-
-    if (!assignment.isNullType() && allocationKey !== null)
-      this.logAssignment(
-        flagKey,
-        allocationKey,
-        assignment,
+    try {
+      const result = this.getAssignmentDetail(
         subjectKey,
-        holdoutKey,
-        holdoutVariation,
+        flagKey,
         subjectAttributes,
+        expectedVariationType,
+        obfuscated,
       );
 
-    return assignment;
+      if (!result.variation) {
+        return defaultValue;
+      }
+
+      return EppoValue.generateEppoValue(result.variation.value, expectedVariationType);
+    } catch (error) {
+      return this.rethrowIfNotGraceful(error, defaultValue);
+    }
   }
 
-  private getAssignmentInternal(
+  public getAssignmentDetail(
```

Comment:
Can make this private for now -- anything in particular you think we should change in this interface?

Reply:
Added docstring instead of making private. I think this function is valuable from a debugging standpoint; we can leave it undocumented for now but can point people to it when trying to help them debug issues.

Reply:
Sounds like a good plan!